hypervisor/src/hypervisor.rs

```diff
@@ -121,10 +121,18 @@ pub trait Hypervisor: Send + Sync {
 ///
 /// Generic MemoryRegion struct
 ///
-pub struct MemoryRegion {
+#[derive(Debug, Default, PartialEq)]
+pub struct UserMemoryRegion {
```

Comment:
Could you please apply the name change in the first commit rather than patching it afterwards.